crs_motion_planning/src/path_planning_utils.cpp

```diff
@@ -109,300 +109,389 @@ bool crsMotionPlanner::generateSurfacePlans(pathPlanningResults::Ptr& results)
+    return false;
+  }
+
+  std::vector<trajectory_msgs::msg::JointTrajectory> second_descartes_trajs;
+  // Combine all rasters together to perform descartes globally
+  if (config_->global_descartes)
+  {
+    std::vector<size_t> traj_lengths;
+    geometry_msgs::msg::PoseArray combined_rasters;
+    for (size_t i = 0; i < split_reachable_rasters.size(); ++i)
+    {
+      traj_lengths.push_back(split_reachable_rasters[i].poses.size());
+      combined_rasters.poses.insert(combined_rasters.poses.end(),
+                                    split_reachable_rasters[i].poses.begin(),
+                                    split_reachable_rasters[i].poses.end());
+    }
+
+    // Perform descartes globally
+    trajectory_msgs::msg::JointTrajectory global_descartes_traj_round1;
+    std::vector<size_t> failed_edges2, failed_vertices2;
+    if (!generateDescartesSeed(combined_rasters, failed_edges2, failed_vertices2, global_descartes_traj_round1))
+    {
+      std::cout << "PREVIOUSLY REACHABLE POINTS ARE NOW UNREACHABLE" << std::endl;
+      results->msg_out = "PREVIOUSLY REACHABLE POINTS ARE NOW UNREACHABLE";
+      return false;
+    }
+
+    // Store global descartes back into individual trajectories
+    if (config_->combine_strips)
+    {
+      split_reachable_rasters.clear();
+      split_reachable_rasters.push_back(combined_rasters);
+      second_descartes_trajs.push_back(global_descartes_traj_round1);
+    }
+    else
+    {
+      size_t start_traj_i = 0;
+      for (size_t i = 0; i < split_reachable_rasters.size(); ++i)
       {
-        split_traj.push_back(joint_traj_msg_out_final);
-        split_reachable_rasters.push_back(std::move(split_strip_ar));
-        any_successes = true;
-        std::cout << "SUCCESS" << std::endl;
+        trajectory_msgs::msg::JointTrajectory curr_global_traj;
+        curr_global_traj.joint_names = global_descartes_traj_round1.joint_names;
+        for (size_t j = 0; j < traj_lengths[i]; ++j)
+        {
+          curr_global_traj.points.push_back(global_descartes_traj_round1.points[start_traj_i + j]);
+        }
+        second_descartes_trajs.push_back(curr_global_traj);
+        start_traj_i += traj_lengths[i];
       }
-      else
+    }
+  }
+  else
+  {
+    for (auto strip : split_reachable_rasters)
+    {
+      trajectory_msgs::msg::JointTrajectory curr_traj;
+      std::vector<size_t> failed_edges2, failed_vertices2;
+      if (!generateDescartesSeed(strip, failed_edges2, failed_vertices2, curr_traj))
       {
-        results->unreachable_waypoints.poses.insert(
-            results->unreachable_waypoints.poses.end(), split_strip_ar.poses.begin(), split_strip_ar.poses.end());
-        std::cout << "ADDED APPROACH AND/OR RETREAT MADE RASTER UNREACHABLE" << std::endl;
+        std::cout << "PREVIOUSLY REACHABLE POINTS ARE NOW UNREACHABLE" << std::endl;
+        results->msg_out = "PREVIOUSLY REACHABLE POINTS ARE NOW UNREACHABLE";
```

Comment:
This seems like a good candidate for `RCLCPP_DEBUG`

Reply:
I went ahead and used RCLCPP_ERROR